crates/cli/commands/src/common.rs

```diff
@@ -199,3 +199,33 @@ impl AccessRights {
 /// [`NodeTypes`](reth_node_builder::NodeTypes) in CLI.
 pub trait CliNodeTypes: NodeTypesWithEngine + NodeTypesForProvider {}
 impl<N> CliNodeTypes for N where N: NodeTypesWithEngine + NodeTypesForProvider {}
+
+/// Helper trait aggregating components required for the CLI.
+pub trait CliNodeComponents<N: CliNodeTypes> {
+    /// Block executor.
+    type Executor: BlockExecutorProvider<Primitives = N::Primitives>;
+    /// Consensus implementation.
+    type Consensus: FullConsensus<N::Primitives, Error = ConsensusError> + Clone + 'static;
+
+    /// Returns the block executor.
+    fn executor(&self) -> &Self::Executor;
+    /// Returns the consensus implementation.
+    fn consensus(&self) -> &Self::Consensus;
```

Comment:
this makes sense, this is likely something we need to integrate in cli trait somehow

cc @fgimenez for vis